pkg/metastore/kv.go

```diff
@@ -14,101 +14,231 @@
 package metastore
 
 import (
+	"crypto/sha512"
```

Comment:
A user has the ability to perform a chosen-plaintext attack to try to generate an ID that is identical to an existing one, thereby potentially leaking metadata of an existing line. It may be overly cautious at this point since we never really said that Parca is multi-tenant, especially not the metastore, but I thought better safe than sorry (therefore I chose the strongest currently available cryptographic hash).